### DISCUSSION TITLE
Rclone setup with backblaze b2

### DISCUSSION THREAD

**Discussion by atwm** [outdoorbits/little-backup-box#294] [Q&A]
I want to backup my photos from an USB source to a cloud target. The target, in this case, is Backblaze B2. The rclone config is working, I can copy or mount the cloud storage and lbb mounts it into `/media/cloud_target`. The problem is that the files must go into the subdirectory named after the bucket I have exposed. 

How can I instruct lbb to write into `/media/cloud_target/MYBUCKETNAME/`and not directly into `/media/cloud_target/`

Thanks

**Comment by outdoorbits** [comment]
Because the cloud services are completely configured by rclone I don't see any clean way to configure a sub folder. Would it be a solution to unser just this folder structure on source?

**Reply by atwm** [reply]
It should be possible to mount the subfolder cleanly. In `lib_storage.py`, the `__mount_cloud` method defines the mount command for rclone as:

```
  Command = f'rclone mount {self.CloudServiceName}: {self.MountPoint} --umask=0 --read-only=false --uid={self.__mount_uid} --gid={self.__mount_gid} --allow-other --config {self.__RCLONE_CONFIG_FILE}'

```

I think the only thing that needs to be changed is to add a variable (via some configuration or environment variable) to do the following:

```
  Command = f'rclone mount {self.CloudServiceName}:{self.BucketName} {self.MountPoint} --umask=0 --read-only=false --uid={self.__mount_uid} --gid={self.__mount_gid} --allow-other --config {self.__RCLONE_CONFIG_FILE}'

```

That is to add a string representing the bucket name after the cloud service name. I assume this should be an empty string by default but should be configurable via the setup page. This should work for all cloud services, as it is the correct rclone syntax. It could also be used in Dropbox for OneDrive to specify the subdirectory to be used for backups.


In the particular case of my b2 bucket:

```
~ % rclone mount b2: /tmp/b2 &
~ % ls -al /tmp/b2/
total 0
drwxr-xr-x   1 andrew  staff    0 Apr 17 22:46 .
drwxrwxrwt  12 root    wheel  384 Apr 17 19:57 ..
drwxr-xr-x   1 andrew  staff    0 Apr 17 22:46 pearshapedbusiness
~ % ls -al /tmp/b2/pearshapedbusiness
total 1
drwxr-xr-x  1 andrew  staff  0 Apr 17 22:46 .
drwxr-xr-x  1 andrew  staff  0 Apr 17 22:46 ..
-rw-r--r--  1 andrew  staff  6 Apr 16 20:32 hello.txt

```

But if I mount it with the subdirectory directly, I can get the correct root directory, which I can write to.
```
% rclone mount b2:pearshapedbusiness /tmp/b2 &
[1] 92639
 ~ % ls -al /tmp/b2/
total 1
drwxr-xr-x   1 andrew  staff    0 Apr 17 22:49 .
drwxrwxrwt  12 root    wheel  384 Apr 17 19:57 ..
-rw-r--r--   1 andrew  staff    6 Apr 16 20:32 hello.txt

```

Setting the subdirectory for cloud services is reasonably achievable. I would really appreciate if this could be considered a possible feature request.

Thanks!

**Comment by outdoorbits** [comment]
Looks good. Please be patient a bit, I'm just working on another feature request. But using the same BucketName for all cloud services is easy to manage.

**Comment by outdoorbits** [comment]
I implemented this. Please configure it in settings/cloud services and please report about your tests.

**Reply by atwm** [reply]
Cool. I’ll try as soon as possible, but probably not before Wednesday 
Thanks!
